src/internal/connector/exchange/service_iterators.go

```diff
@@ -334,153 +345,203 @@ func FetchEventIDsFromCalendar(
 	}
 
 	// Events don't have a delta endpoint so just return an empty string.
-	return ids, "", errs.ErrorOrNil()
+	return ids, deltaUpdate{}, errs.ErrorOrNil()
 }
 
+// ---------------------------------------------------------------------------
+// contacts
+// ---------------------------------------------------------------------------
+
 // FetchContactIDsFromDirectory function that returns a list of  all the m365IDs of the contacts
 // of the targeted directory
 func FetchContactIDsFromDirectory(
 	ctx context.Context,
 	gs graph.Servicer,
 	user, directoryID, oldDelta string,
-) ([]string, string, error) {
+) ([]string, deltaUpdate, error) {
 	var (
-		errs     *multierror.Error
-		ids      []string
-		deltaURL string
+		errs       *multierror.Error
+		ids        []string
+		deltaURL   string
+		resetDelta bool
 	)
 
-	options, err := optionsForContactFoldersItemDelta([]string{"parentFolderId"})
+	options, err := optionsForContactFoldersItemDelta([]string{"isRead"})
 	if err != nil {
-		return nil, deltaURL, errors.Wrap(err, "getting query options")
+		return nil, deltaUpdate{}, errors.Wrap(err, "getting query options")
 	}
 
-	builder := gs.Client().
-		UsersById(user).
-		ContactFoldersById(directoryID).
-		Contacts().
-		Delta()
+	getIDs := func(builder *msuser.ItemContactFoldersItemContactsDeltaRequestBuilder) error {
+		for {
```

Comment:
same loop as below, just moved up to an ad-hoc func so that it can be called twice (in case of delta invalidation) without code duplication.

Reply:
thanks for the callout